applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -687,12 +690,15 @@ class EigensolverStrategy
             // }
 
             // fill the EIGENVECTOR_MATRIX
-            for (std::size_t i = 0; i < NumEigenvalues; i++)
+            for (std::size_t i = 0; i < NumEigenvalues; i++) {
                 for (std::size_t j = 0; j < NumNodeDofs; j++)
                 {
                     auto itDof = std::begin(NodeDofs) + j;
-                    rNodeEigenvectors(i,j) = rEigenvectors(i,(*itDof)->EquationId());
+                    if ((*itDof)->IsFree()) {
```

Comment:
this is required for the EliminationB&S because the solution (`rEigenvectors`) does not contain the fixed dofs
Guess none tried this before :)

Reply:
now this is, where this segmentation fault came from...

Reply:
MINOR: this loop should be made openmp parallel...

Reply:
If we want to do this then we have to refactor the [entire loop](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp#L671-L697) bcs of the allocation in parallel of the DataValueContainer [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp#L675)
I can do it but only after we converged on how to proceed with this

Reply:
there is another problem in this line:
For cases with mixed elements (e.g. trusses & beams) on nodes that belong only to trusses we also have the `ROTATION` dofs. These dofs are not in the `DofSet` in the B&S, but they are still there. Meaning that if we loop the dofs (which we are doing here) then we also get those "inactive" dofs. They return the default equation Id which is 0 and leads to wrong results.
I don't know to differentiate the "active" from the "inactive" dofs. We thought of setting an equation Id of -1 but this does not work bcs it is a `std::size_t`

Maybe someone has an idea?